### PR TITLE
fix: edict of nerotaurus coming in and out of play

### PR DESCRIPTION
**Pull request by dmikalova**
fix: Rewrote Edict of Nerotaurus to handle coming in and out of play, being blanked, and coming in mid-reap with multiple effects, fixing #4708 
feat: added tests similar tests for Snag's Mirror and Etan's Jar being blanked

Tried updating the original code, which became more and more convoluted with each corner case. Tried using Snag's Mirror reaction method, which set me on the right track, but needed to use Etan's Jar's lastingEffect instead of persistentEffect to handle being blanked.

#4708

### CODE REVIEW

File: server/game/cards/06-WoE/EdictOfNerotaurus.js [hunk 1]
```diff
@@ -1,59 +1,38 @@
...
+            gameAction: ability.actions.lastingEffect(() => ({
+                until: {
+                    onUseCard: (event) => event.card.type === 'creature',
+                    onTurnEnd: () => true
+                },
                 targetController: 'any',
-                condition: () => this.lastCreatureFought,
                 effect: ability.effects.cardCannot('fight')
-            })
+            }))
```
Comment (fionawhim) [inline server/game/cards/06-WoE/EdictOfNerotaurus.js:33]: Oh nice! This is really elegant, and feels like a perfect translation of the card text.

Comment (dmikalova) [reply server/game/cards/06-WoE/EdictOfNerotaurus.js:33]: Thanks! Yeah it really is exactly what I would have hoped for
